### PR TITLE
fix: router output ordering for tiktok_ads

### PR DESCRIPTION
**Pull request by maheshkutty**
## What are the changes introduced in this PR?

This pull request introduces a new utility function, `sortBatchesByMinJobId`, to ensure consistent ordering of batched requests based on the minimum `jobId` in their metadata. The function is integrated into multiple files, enhancing batch processing logic for TikTok Ads transformations and other utilities. Additionally, comprehensive unit tests have been added to validate its behavior across various scenarios.

### Enhancements to batch processing:

* **Utility function addition:** Added `sortBatchesByMinJobId` in `src/v0/util/index.js` to sort batches by the minimum `jobId` in their metadata, ensuring consistent ordering. This replaces inline sorting logic in multiple places. [[1]](diffhunk://#diff-f950b7c48fcbbbb1f78c6542a506d131d0a9cb25ca6ad3d6d85683bdc4770ff7L2216-R2242) [[2]](diffhunk://#diff-f950b7c48fcbbbb1f78c6542a506d131d0a9cb25ca6ad3d6d85683bdc4770ff7L2269-R2291) [[3]](diffhunk://#diff-f950b7c48fcbbbb1f78c6542a506d131d0a9cb25ca6ad3d6d85683bdc4770ff7R2549)
* **Integration in TikTok Ads transformations:** Updated `processRouterDest` in both `src/v0/destinations/tiktok_ads/transform.js` and `src/v0/destinations/tiktok_ads/transformV2.js` to use `sortBatchesByMinJobId` for sorting batched responses. [[1]](diffhunk://#diff-5953a19c81563325c0f7fb2ebcec30bde82a703c62a4aedec2ed97d027d984b5L285-R286) [[2]](diffhunk://#diff-c5e1e1a72f3a22947dd8529c1d624acaf16972e0278aa3efbd2e189e534e7d88L760-R763)

### Testing improvements:

* **Unit tests for `sortBatchesByMinJobId`:** Added extensive test cases in `src/v0/util/index.test.js` to validate sorting functionality, including edge cases like empty arrays, single metadata items, and already sorted inputs.

### Code cleanup:

* **Removed redundant logic:** Eliminated inline sorting code in `combineBatchRequestsWithSameJobIds` and replaced it with the new utility function for better reusability and maintainability.

### Other changes:

* **TikTok Ads integration test adjustment:** Removed TikTok Ads-specific exclusion in `test/integrations/component.test.ts`, likely as part of broader test cleanup or updates.

## What is the related Linear task?

Resolves INT-3733

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4364/index.html)

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4364/index.html)

File: src/v0/util/index.test.js [hunk 1]
```diff
@@ -1406,6 +1407,84 @@ describe('getBodyFromV2SpecPayload', () => {
...
+      expect(sortBatchesByMinJobId(input)).toEqual(expected);
+    });
+
+    it('should handle batches with multiple metadata items and unordered jobIds', () => {
+      const input = [
+        { metadata: [{ jobId: 8 }, { jobId: 3 }] },
+        { metadata: [{ jobId: 2 }, { jobId: 7 }] },
+        { metadata: [{ jobId: 5 }, { jobId: 9 }] },
+      ];
+      const expected = [
```
Comment (ItsSudip) [inline src/v0/util/index.test.js:1446]: We can remove this as it is not used.

**Review by ItsSudip — DISMISSED** [review]

**Review by ItsSudip — DISMISSED** [review]

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4364/index.html)

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4364/index.html)

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4364/index.html)
